core/modules/field/field.block.inc

```diff
@@ -80,9 +80,34 @@ class FieldBlock extends Block {
     list($entity_type, $field_name) = explode('-', $this->childDelta);
     $entity = $this->contexts[$entity_type]->data;
 
-    // Do not render if the entity is not found on this layout.
-    if (empty($entity)) {
-      return NULL;
+    // Do not proceed if the entity is not an EntityInterface object
+    if (!is_a($entity, 'EntityInterface')) {
+      if ($entity_type == 'user') {
+        if (isset($entity->uid)) {
+          // Get a complete user entity.
+          $context_info = layout_get_context_info('user');
+          // Check if a load function has been set by some other module. If not, default to user_load().
+          if (isset($context_info['load callback']) && function_exists($context_info['load callback'])) {
+            $load_function = $context_info['load callback'];
+          }
+          else {
+            $load_function = 'user_load';
+          }
+          $entity = $load_function($entity->uid);
+          if (is_a($entity, 'EntityInterface')) {
+            $this->contexts[$entity_type]->setData($entity);
+          }
+          else {
+            return NULL;
+          }
+        }
+        else {
+          return NULL;
+        }
+      }
+      else {
+        return NULL;
+      }
```

Comment:
The number of nested conditionals here is confusing to follow. I suggest a flatter structure and only need to escape once at the end if the entity is still not a full entity:

```suggestion
    // The user entity needs to implement EntityInterface in order for all the
    // user fields to be loaded.
    // See https://github.com/backdrop/backdrop-issues/issues/2398.
    if ($entity_type == 'user' && !is_a($entity, 'EntityInterface')) {
      // Only attempt if not Anonymous.
      if (isset($entity->uid)) {
        $context_info = layout_get_context_info('user');
        // Check if a load function has been set by some other module. If not,
        // default to user_load().
        if (isset($context_info['load callback']) && function_exists($context_info['load callback'])) {
          $load_function = $context_info['load callback'];
        }
        else {
          $load_function = 'user_load';
        }
        $entity = $load_function($entity->uid);
        if (is_a($entity, 'EntityInterface')) {
          $this->contexts[$entity_type]->setData($entity);
        }
      }
    }

    // Do not proceed if the entity still doesn't implement EntityInterface.
    if (!is_a($entity, 'EntityInterface')) {
      return NULL;
    } 
```

Reply:
Something I wasn't sure about was what kind of error logging should be part of this. Is it OK to just fail out of the function? Or should we be tracking the reasons?

Reply:
That's why I was distinguishing all the various ways it might fail.